CMakeLists.txt

```diff
@@ -382,27 +381,26 @@ endfunction()
 # Define which library variants to build and which flags to use.
 # The order is <arch> <name suffix> <flags>
 add_library_variants(aarch64-none-elf picolibc
-    aarch64         ""                  "-march=armv8-a"
+    aarch64         ""                  "-march=armv8-a"                                        "${picolibc_SOURCE_DIR}/scripts/run-aarch64"
 )
 add_library_variants(arm-none-eabi picolibc
-    armv6m          soft_nofp           "-mfloat-abi=soft -march=armv6m"
-    armv7em         hard_fpv4_sp_d16    "-mfloat-abi=hard -march=armv7em -mfpu=fpv4-sp-d16"
-    armv7em         hard_fpv5_d16       "-mfloat-abi=hard -march=armv7em -mfpu=fpv5-d16"
-    armv7em         soft_nofp           "-mfloat-abi=soft -march=armv7em -mfpu=none"
-    armv7m          soft_nofp           "-mfloat-abi=soft -march=armv7m+nofp"
-    armv8m.main     hard_fp             "-mfloat-abi=hard -march=armv8m.main+fp"
-    armv8m.main     soft_nofp           "-mfloat-abi=soft -march=armv8m.main+nofp"
-    armv8.1m.main   hard_fp             "-mfloat-abi=hard -march=armv8.1m.main+fp"
-    armv8.1m.main   hard_nofp_mve       "-mfloat-abi=hard -march=armv8.1m.main+nofp+mve"
-    armv8.1m.main   soft_nofp_nomve     "-mfloat-abi=soft -march=armv8.1m.main+nofp+nomve"
+    armv6m          soft_nofp           "-mfloat-abi=soft -march=armv6m"                        "${picolibc_SOURCE_DIR}/scripts/run-thumbv6m"
+    armv7em         hard_fpv4_sp_d16    "-mfloat-abi=hard -march=armv7em -mfpu=fpv4-sp-d16"     "${picolibc_SOURCE_DIR}/scripts/run-thumbv7m"
+    armv7em         hard_fpv5_d16       "-mfloat-abi=hard -march=armv7em -mfpu=fpv5-d16"        "${picolibc_SOURCE_DIR}/scripts/run-thumbv7m"
+    armv7em         soft_nofp           "-mfloat-abi=soft -march=armv7em -mfpu=none"            "${picolibc_SOURCE_DIR}/scripts/run-thumbv7m"
+    armv7m          soft_nofp           "-mfloat-abi=soft -march=armv7m+nofp"                   "${picolibc_SOURCE_DIR}/scripts/run-thumbv7m"
+    armv8m.main     hard_fp             "-mfloat-abi=hard -march=armv8m.main+fp"                "${picolibc_SOURCE_DIR}/scripts/run-arm"
```

Comment:
Does run-arm need extra arguments to select the sub-architecture?

Reply:
It's pretty complex script that detects and then sets variety of qemu parameters. That might be the reason it works for amrv7em_soft_nofp and for others it sets wrong parameters.